### ISSUE TITLE
SVTR Text Recognition: Zero Accuracy Issue & Need Guidance for Custom Pipeline

### ISSUE THREAD

**Issue by btwitsayush** [PaddlePaddle/PaddleOCR#15295]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### **Description**
I'm new to PaddleOCR and currently building a custom text recognition pipeline for address labels and tracking numbers. While implementing this, I encountered multiple issues—most critically, the SVTR model always trains to 0% accuracy. I'm seeking help with:

- Resolving these training/export errors
- Getting guidance to correctly set up a custom text recognition pipeline

### **Current Issues**

1. SVTR model trains but stays at 0% accuracy
2. CRNN model attempts lead to segmentation faults
3. Exporting fails with SVTRRecResizeImg error

### **Environment**

- PaddlePaddle version: 2.6.2
- Running on: Google Colab (Linux)
- GPU: Available, CUDA enabled

### **Dataset Details**

- Total images: 50
- Training: 45
- Testing: 5
- Image format: BGR, 3x32x320
- Unique characters: 63
- Content: Address labels and tracking numbers (data is encrypted/masked for privacy)

### **Data Formats:**

**- Address Labels:**

Name XXXXX Address XXXXX Ltd.
Floor XX, Building XXX
City - XXXXXX
Phone XXXXXXXXXX Email Pincode XXXXXX

**Tracking Numbers:**

XXXXXXXXXXXXX (12–15 digit numbers)

### **Model Configuration (YAML)**

**Architecture:**
  model_type: rec
  algorithm: SVTR
**Backbone:**
  name: MobileNetV1Enhance
  scale: 0.5
**Neck:**
  name: SequenceEncoder
  encoder_type: svtr
  dims: 64
  depth: 2
  hidden_dims: 120
  use_guide: True
**Head:**
  name: CTCHead
  fc_decay: 0.00001

### **Training Results**

- Epochs: 20
- Best Epoch: 19
- Accuracy: 0.0
- Normalized Edit Distance: 0.06458
- Loss: From 32.9 to 31.0
- Learning Rate: 5e-6
- Pretrained Model: ch_PP-OCRv3_rec_train
- Batch Size: 4
- Max Text Length: 500

### **Questions**

1. Why is accuracy stuck at 0?
2. Is learning rate 5e-6 too low for fine-tuning?
3. How to fix the SVTRRecResizeImg export error?
4. What is the correct pipeline for training with a small, custom dataset?

### **Pipeline Help Needed**

**Please guide me on:**

- Preparing a custom dataset properly (format, label structure, etc.)
- Generating and validating the dict.txt (character dictionary)
- Adjusting training config for small datasets
- Full steps to set up a working custom text recognition pipeline
- Choosing the right architecture (SVTR, CRNN, etc.) for small  & large datasets with limited labeled images

### **Additional Notes**

- Dataset contains sensitive personal information, hence encrypted.
- Custom pipeline is for logistics use cases (address & tracking number recognition).
- Willing to contribute findings back to the community.

### **Request for Assistance**

1. Help me fix the SVTR training and export issues
2. Share a working config or example for small-scale text recognition training
3. Recommend best practices for datasets with private content
4. Guide me in building a robust custom recognition pipeline

/label ~bug ~text-recognition ~training ~help-wanted ~documentation
/cc @PaddleOCR-maintainers

### 🏃‍♂️ Environment (运行环境)

Google Colab : Linux

### **Config which I used for training**

rec_config_content = f'''
Global: 
  debug: true 
  use_gpu: true
  epoch_num: 20
  log_smooth_window: 20
  print_batch_step: 5
  save_model_dir: ./output/rec_finetune
  save_epoch_step: 2
  eval_batch_step: [0, 50]
  cal_metric_during_train: true
  pretrained_model: ./pretrained_models/ch_PP-OCRv3_rec_train/best_accuracy.pdparams
  checkpoints: null
  save_inference_dir: ./inference/rec_finetune
  use_visualdl: false
  character_dict_path: {dict_path}
  max_text_length: 500
  character_type: ch
  use_space_char: true
  save_res_path: ./output/rec/predicts_rec.txt

Architecture:
  model_type: rec
  algorithm: SVTR
  Transform: null
  Backbone:
    name: MobileNetV1Enhance
    scale: 0.5
    last_conv_stride: [1, 2]
    last_pool_type: avg
  Neck:
    name: SequenceEncoder
    encoder_type: svtr
    dims: 64
    depth: 2
    hidden_dims: 120
    use_guide: True
  Head:
    name: CTCHead
    fc_decay: 0.00001

Loss:
  name: CTCLoss

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.000005
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 0.000005

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: {TRAIN_IMAGES_DIR}
    label_file_list: ["{TRAIN_LABEL_FILE}"]
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: false
      - RecAug:
      - CTCLabelEncode:
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ["image", "label", "length"]
  loader:
    shuffle: true
    batch_size_per_card: 4
    drop_last: true
    num_workers: 2

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: {TEST_IMAGES_DIR}
    label_file_list: ["{TEST_LABEL_FILE}"]
    transforms:
      - DecodeImage:
          img_mode: BGR
          channel_first: false
      - CTCLabelEncode:
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ["image", "label", "length"]
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 4
    num_workers: 2
'''

**Comment by cuicheng01** [comment]
Thank you for using PaddleOCR. The reason for your issue is that your data volume is too small. To train a model with reasonable accuracy, you may need at least several thousand images, or even tens of thousands.

**Comment by TingquanGao** [comment]
The issue has no response for a long time and will be closed. You can reopen or new another issue if are still confused.

---
_From Bot_
